src/crappy/actuator/phidgets_stepper4a.py

```diff
@@ -304,4 +306,7 @@ def _on_position_change(self, _: Stepper, position: float) -> None:
   def _on_end(self, _: DigitalInput, __) -> None:
     """Callback when a switch is hit."""
 
-    self.stop()
+    for switch in self._switches:
```

Comment:
The callback is tied to a single switch, but here you're checking all the switches at once. This is not the intended behavior.
also, the new state of the switch is available in one of the unused arguments of the method.